### PR TITLE
OWTimeSlice: Handle data with one time point

### LINKED ISSUE #126: Timeseries: handle no time delta
##### Timeseries version

0.3.7

##### Orange version

3.27.dev

##### Expected behavior
Constant data doesn't cause issues.

##### Actual behavior
Timeseries' make_timeseries fails when no delta in the data (all dates are the same).

##### Steps to reproduce the behavior
Data:

Date|Value
----|----
2020-01-01|1
2020-01-01|2
2020-01-01|3
2020-01-01|4
2020-01-01|5

Connect to As Timeseries or Line Chart.

##### Additional info (worksheets, data, screenshots, ...)
Stack trace:
```
Traceback (most recent call last):
  File "/Users/ajda/opt/miniconda3/envs/o3/lib/python3.8/site-packages/orangecanvas/scheme/signalmanager.py", line 936, in __process_next
    if self.__process_next_helper(use_max_active=True):
  File "/Users/ajda/opt/miniconda3/envs/o3/lib/python3.8/site-packages/orangecanvas/scheme/signalmanager.py", line 974, in __process_next_helper
    self.process_node(selected_node)
  File "/Users/ajda/opt/miniconda3/envs/o3/lib/python3.8/site-packages/orangecanvas/scheme/signalmanager.py", line 605, in process_node
    self.send_to_node(node, signals_in)
  File "/Users/ajda/orange/orange-widget-base/orangewidget/workflow/widgetsscheme.py", line 792, in send_to_node
    self.process_signals_for_widget(node, widget, signals)
  File "/Users/ajda/orange/orange-widget-base/orangewidget/workflow/widgetsscheme.py", line 826, in process_signals_for_widget
    handler(*args)
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owtabletotimeseries.py", line 75, in set_data
    self.on_changed()
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owtabletotimeseries.py", line 78, in on_changed
    self.commit()
  File "/Users/ajda/orange/orange-widget-base/orangewidget/gui.py", line 1590, in unconditional_commit
    do_commit()
  File "/Users/ajda/orange/orange-widget-base/orangewidget/gui.py", line 1598, in do_commit
    commit()
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/widgets/owtabletotimeseries.py", line 90, in commit
    ts = Timeseries.make_timeseries_from_continuous_var(data,
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/timeseries.py", line 198, in make_timeseries_from_continuous_var
    ts.time_variable = time_var
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/timeseries.py", line 227, in time_variable
    self.time_delta = TimeDelta(self.time_values)
  File "/Users/ajda/orange/orange3-timeseries/orangecontrib/timeseries/timeseries.py", line 49, in __init__
    self.min = deltas[0]
IndexError: list index out of range
```

Issue present for first data set in https://github.com/biolab/orange3-timeseries/issues/115.

### PR DESCRIPTION
**Pull request by irgolic**
Merge after #125.

##### Issue
Fixes #126 

##### Includes
- [X] Code changes
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: orangecontrib/timeseries/timeseries.py [hunk 1]
```diff
@@ -22,13 +22,23 @@ def __init__(self, time_values):
...
 
-        deltas = list(np.sort(np.unique(np.diff(self.time_values))))
+        deltas = list(np.sort(np.unique(np.diff(np.sort(self.time_values)))))
+
+        # in case several rows fall on the same datetime, remove the zero
+        if deltas[0] == 0:
+            if len(deltas) == 1:
+                no_delta()
+                return
+            deltas = deltas[1:]
```
Comment (ajdapretnar) [inline orangecontrib/timeseries/timeseries.py:41]: `deltas = deltas[1:] `could also result in `len(deltas) == 1`, but wouldn't be checked.

```
deltas = list(np.sort(np.unique(np.diff(np.sort(self.time_values)))))
if deltas and deltas[0] == 0:
    deltas.pop(0)
if len(deltas) <= 1:
    no_deltas()
```

I think it should do the same thing, but cover all the cases.

Comment (ajdapretnar) [inline orangecontrib/timeseries/timeseries.py:41]:
```
deltas = list(np.sort(np.unique(np.diff(np.sort(self.time_values)))))
if deltas and deltas[0] == 0:
    deltas.pop(0)
if not deltas:
    no_deltas()
```

I think it should do the same thing, but is shorter?

Comment (kernc) [reply orangecontrib/timeseries/timeseries.py:41]: `np.unique()` returns already sorted unique elements.

Comment (irgolic) [reply orangecontrib/timeseries/timeseries.py]: You're right, that's much clearer. Thanks, both!

**Comment by ajdapretnar** [comment]
That's actually really nice! 👍

**Pull request merged**